src/printer.js

```diff
@@ -1107,20 +1110,70 @@ function printClass(path, options, print) {
     declaration.push(printArgumentsList(path, options, print));
   }
 
-  const partsDeclarationGroup = [];
-
-  if (node.extends) {
-    partsDeclarationGroup.push(printClassPart(path, options, print, "extends"));
-  }
-
-  if (node.implements) {
-    partsDeclarationGroup.push(
-      printClassPart(path, options, print, "implements")
+  // Only `class` can have `extends` and `implements`
+  if (node.extends && node.implements) {
+    declaration.push(
+      conditionalGroup(
+        [
+          concat([
+            printClassPart(path, options, print, "extends", " ", " "),
+            printClassPart(path, options, print, "implements", " ", " ")
```

Comment:
Maybe better not to pass last two arguments if they match the default?

Reply:
:+1: